File: 2/dp/MinCostStairs.java


public class MinCostStairs {

    public static void main(String[] args) {
        int[] arr = {1,100,1,1,1,100,1,1,100,1};
        int cost = 0;
        for(int i=2;i<arr.length;i++){
            arr[i] = arr[i]+Math.min(arr[i-1],arr[i-2]);
        }
        cost = Math.min(arr[arr.length-1],arr[arr.length-2]);
        System.out.println(cost);
    }


}

File: 1/Leetcode/dp/min_cost_stairs.cxx
# include<stdio.h> 
#include<iostream>
#define min(a,b) ((a<b)?a:b)
using namespace std;

int arr[100];

int climbStairs(int n) {
        
        arr[n-1]=climbStairs(n-1);
        arr[n-2]=climbStairs(n-2);
        arr[n]=min(arr[n-1],arr[n-2]);
        return arr[n];
   }

int main() 
{ 
	int n;
	cin>>n; 
	cout<<"ans:"<<climbStairs(n)<<endl;
	for(int i=0;i<10;i++)
	cout<<arr[i]<<" ";
	return 0; 
} 

